src/pybind/mgr/telemetry/module.py

```diff
@@ -327,6 +331,110 @@ def gather_crashinfo(self) -> List[Dict[str, str]]:
+                perf_schema[daemon_type] = initial_info[daemon]
+            else:
+                for collection in perf_schema[daemon_type]:
+                    for field in perf_schema[daemon_type][collection]:
+                        if type(perf_schema[daemon_type][collection][field]) != str:
+                            perf_schema[daemon_type][collection][field] += \
+                                    initial_info[daemon][collection][field]
+
+        # Initialize 'result' dict
+        result: Dict[str, dict] = defaultdict(dict)
+        for daemon in perf_schema:
+            result[daemon] = {}
+            for collection in perf_schema[daemon]:
+                col_split = collection.split('.')
+                result[daemon][col_split[0]] = {}
+
+        # Derive 'perf dump' from the output of 'perf schema'
+        for daemon in perf_schema:
+            for collection in perf_schema[daemon]:
+
+                # Split the collection to avoid redundancy in json report; i.e.:
+                #   bluestore.kv_flush_lat, bluestore.kv_final_lat --> 
+                #   bluestore: kv_flush_lat, kv_final_lat
+                col_split = collection.split('.')
+
+                # Initialize the collection dictionary; i.e.:
+                #   "bluestore": {
+                #       "kv_flush_lat": {
+                #       },
+                #   },
+                result[daemon][col_split[0]][col_split[1]] = \
+                        result[daemon][col_split[0]].get(col_split[1], {})
+
+                # Derive sum from perf_schema output; applies in all situations.
+                #   'sum' = the perf_schema 'value' field
+                # (see https://docs.ceph.com/en/latest/dev/perf_counters/)
+                _sum = perf_schema[daemon][collection]['value']
+
+                # In the case that there is a 'count' field in perf_schema, derive 
+                # 'avgcount' and 'avgtime' from perf_schema output.
+                #
+                #   'avgcount' = the perf_schema 'count' field
+                #   'avgtime' = 'sum' / 'avgcount'
+                # (see https://docs.ceph.com/en/latest/dev/perf_counters/)
+                #
+                # Example of perf_schema output:
+                #   "mon.b": {
+                #       "bluestore.kv_flush_lat": {
+                #           "count": 2431,
+                #           "description": "Average kv_thread flush latency",
+                #           "nick": "fl_l",
+                #           "priority": 8,
+                #           "type": 5,
+                #           "units": 1,
+                #           "value": 88814109
+                #       },
+                #   },
+                if 'count' in perf_schema[daemon][collection]:
+                    avgcount = perf_schema[daemon][collection]['count']
+
+                    # Check for divide by 0
+                    if avgcount != 0:
+                        _sum = _sum / 1000000000
```

Comment:
This could be wrong, but my logic here was to compare the output from self.get_all_perf_counters() to perf dump output. If you compare, for instance, the output for 'bluestore.kv_flush_lat' from self.get_all_perf_counters() to the same counter's output from running `ceph tell osd.0 perf dump`, you get:

1. self.get_all_perf_counters() output:
```
"bluestore.kv_flush_lat": {
                "count": 2409,
                "description": "Average kv_thread flush latency",
                "nick": "fl_l",
                "priority": 8,
                "type": 5,
                "units": 1,
                "value": 91315021
            },
```
2. `ceph tell osd.0 perf dump` output:
```
"bluestore": {
        "kv_flush_lat": {
            "avgcount": 2527,
            "sum": 0.091114205,
            "avgtime": 0.000036056
        },
```
The 'value' field corresponds with 'sum', but the implementation of perf dump shows that the sum as been divided by 1000000000. This only seems to occur in perf dump when there is a 'count' field in the corresponding self.get_all_perf_counters() output.

If there is no 'count' field present for a counter in self.get_all_perf_counters(), the sum remains as-is. For instance:

1. self.get_all_perf_counters() output for the 'bluestore-pricache:data.pri1_bytes' counter:
```
"bluestore-pricache:data": {
        "pri1_bytes": 8192,
```
2. `ceph tell osd.0 perf dump` output for the 'bluestore-pricache:data.pri1_bytes' counter:
```
"bluestore-pricache:data.pri1_bytes": {
                "description": "bytes allocated to pri1",
                "nick": "p1",
                "priority": 5,
                "type": 2,
                "units": 0,
                "value": 8192
            },
```
I believe that implementation is happening for perf dump in common/perf_counters.cc in these lines: 
https://github.com/ceph/ceph/blob/0509deb6a895a98e3e582cbb849606bc559b963c/src/common/perf_counters.cc#L422-L439
```
      if (d->type & PERFCOUNTER_LONGRUNAVG) {
	f->open_object_section(d->name);
	pair<uint64_t,uint64_t> a = d->read_avg();
	if (d->type & PERFCOUNTER_U64) {
	  f->dump_unsigned("avgcount", a.second);
	  f->dump_unsigned("sum", a.first);
	} else if (d->type & PERFCOUNTER_TIME) {
	  f->dump_unsigned("avgcount", a.second);
	  f->dump_format_unquoted("sum", "%" PRId64 ".%09" PRId64,
				  a.first / 1000000000ull,
				  a.first % 1000000000ull);
          uint64_t count = a.second;
          uint64_t sum_ns = a.first;
          if (count) {
            uint64_t avg_ns = sum_ns / count;
            f->dump_format_unquoted("avgtime", "%" PRId64 ".%09" PRId64,
                                    avg_ns / 1000000000ull,
                                    avg_ns % 1000000000ull);
```